### ISSUE TITLE
Question about validation

### ISSUE THREAD

**Issue by soon-will** [jiangqy/DPSH-pytorch#3]
Hi, Jang.

As we can see, you test the model after one epoch using test set as queries to retrieval from train set. So, I have some trouble about that.

1. It seems that you save the model after all train are finished, then how do you ensure the last model is the best one(lowerst loss or highest mAP)?

2. Why do you use mAP  to  val the model by using test set as queries to retrieval from train set? How about use the loss to val the model?

Thanks for your reply.

**Comment by jiangqy** [comment]
this testing is just used for debugging. For data splitting, I follow the setting of the paper. If you want to  perform cross-validation, you can split the dataset by yourself.
